src/main/java/org/apache/sysds/runtime/matrix/data/MatrixBlock.java

```diff
@@ -2764,7 +2780,15 @@ else if(!sparse && !isEmptyBlock(false)
 			//note: we apply multi-threading in a best-effort manner here
 			//only for expensive operators such as exp, log, sigmoid, because
 			//otherwise allocation, read and write anyway dominates
-			ret.allocateDenseBlock(false);
+			if (!op.isInplace() || this.isEmpty())
+			{	//allocate dense output block
+				ret.allocateDenseBlock(false);
+			}
+			else
+			{
+				ret = this;
+			}
+			//ret.allocateDenseBlock(false);
```

Comment:
formatting. 

- { should be same line as statement
- if there is only one line in the if or else statement then remove "{" "}" (the case here)
- Remove commented code that is not used.